src/Stryker.Core/Stryker.Core/Reporters/Json/SourceFiles/SourceFile.cs

```diff
@@ -0,0 +1,44 @@
+using Microsoft.Extensions.Logging;
+using Stryker.Core.Logging;
+using Stryker.Core.ProjectComponents;
+using System.Collections.Generic;
+
+namespace Stryker.Core.Reporters.Json.SourceFiles
+{
+    public class SourceFile
+    {
+        public string Language { get; init; }
+        public string Source { get; init; }
+        public ISet<JsonMutant> Mutants { get; init; }
+
+        public SourceFile()
+        {
+        }
+
+        public SourceFile(ReadOnlyFileLeaf file, ILogger logger = null)
+        {
+            logger ??= ApplicationLogging.LoggerFactory.CreateLogger<SourceFile>();
```

Comment:
Why is the logger created here and not in constructor?

Reply:
This is the constructor? Also this was a rename, it has always been like this.

Reply:
Whoops, didn't pay attention sorry